app/etc/di.xml

```diff
@@ -1840,4 +1840,5 @@
             </argument>
         </arguments>
     </type>
+    <type name="Magento\Framework\Indexer\CacheContext" shared="false" />
```

Comment:
Isn't it enough? should we really change `get` to `create` in all places?

Reply:
https://github.com/magento/magento2/blob/53bcc78df1bfb8fc8335e98f87fff37e227813cd/lib/internal/Magento/Framework/ObjectManager/ObjectManager.php#L65-L73

I don't think that `ObjectManager->get` checks for shared instances. Rather, in the di process it will call either get or create based on whether it's shared.